src/net_processing.cpp

```diff
@@ -3474,7 +3474,11 @@ bool PeerLogicValidation::SendMessages(CNode* pto, std::atomic<bool>& interruptM
             if (pto->nNextInvSend < nNow) {
                 fSendTrickle = true;
                 // Use half the delay for outbound peers, as there is less privacy concern for them.
-                pto->nNextInvSend = PoissonNextSend(nNow, INVENTORY_BROADCAST_INTERVAL >> !pto->fInbound);
+                if (pto->fInbound) {
```

Comment:
For the sake of encapsulation, I think you can put more of this logic into `int64_t CConnman::PoissonNextSendTo(int64_t now, CNode* node)`, including the check on `fInbound` and the half delay comment.

Reply:
@jimpo Does this comment still apply? `CConnman::PoissonNextSendTo` doesn't take a `CNode*` argument anymore, and I think it's better separation to keep the logic for determining the interval separate from the logic for computing the next send.

Reply:
I'd still prefer a separate method for these four lines `CConnman::PoissonNextSendTo(int64_t now, CNode* node)`. Certainly having the separate `PoissonNextSend` which actually computes the interval is a good idea (which already exists)`, but I think `CConnman::PoissonNextSendTo` as defined currently just does the interval caching by netgroup. I find the interface weird and I don't understand the split between logic there and here.

That said, it's a pretty minor point and I don't feel too strongly.